lib/src/main/java/com/orange/ods/compose/component/navigationdrawer/OdsModalDrawer.kt

```diff
@@ -100,169 +103,274 @@ fun OdsModalDrawer(
     )
 }
 
-sealed class OdsModalDrawerItem
-data class OdsModalDrawerSectionLabel(val label: String) : OdsModalDrawerItem()
-object OdsModalDrawerDivider : OdsModalDrawerItem()
-data class OdsModalDrawerListItem(@DrawableRes val icon: Int?, val text: String) : OdsModalDrawerItem()
+/**
+ * Represents an item in an [OdsModalDrawer] content.
+ * It can be a clickable list item ([OdsModalDrawerListItem]), a section label ([OdsModalDrawerSectionLabel]) or
+ * a divider ([OdsModalDrawerDivider]).
+ * These items will be displayed vertically in the [OdsModalDrawer] after the header part.
+ */
+sealed class OdsModalDrawerItem : OdsComponentContent<OdsModalDrawerItem.ExtraParameters>() {
+    data class ExtraParameters(val selected: Boolean, val onClick: (OdsModalDrawerListItem) -> Unit) : OdsComponentContent.ExtraParameters()
+}
 
-@Composable
-private fun ModalDrawerItem(item: OdsModalDrawerItem, selected: Boolean, onClick: (OdsModalDrawerListItem) -> Unit) {
-    return when (item) {
-        is OdsModalDrawerSectionLabel -> {
-            Column {
-                OdsDivider()
-                OdsTextBody2(
-                    modifier = Modifier.padding(top = dimensionResource(id = R.dimen.spacing_m), start = dimensionResource(id = R.dimen.spacing_m)),
-                    text = item.label
-                )
-            }
-        }
-        is OdsModalDrawerListItem -> {
-            CompositionLocalProvider(LocalRippleTheme provides OdsModalDrawerListItemRippleTheme) {
-                OdsListItem(
-                    text = item.text,
-                    textColor = if (selected) OdsTheme.colors.primaryVariant else OdsTheme.colors.onSurface,
-                    textStyle = if (selected) OdsTheme.typography.subtitle2 else OdsTheme.typography.subtitle2.copy(fontWeight = FontWeight.Bold),
-                    modifier = Modifier
-                        .selectable(selected = selected, onClick = { onClick(item) })
-                        .let {
-                            if (selected) it.background(OdsTheme.colors.primaryVariant.copy(alpha = SelectedItemOpacity)) else it
-                        },
-                    icon = item.icon?.let {
-                        OdsListItemIcon(
-                            painterResource(id = it),
-                            "",
-                            if (selected) OdsTheme.colors.primaryVariant else OdsTheme.colors.onSurface
-                        )
-                    }
-                )
-            }
-        }
-        is OdsModalDrawerDivider -> {
+/**
+ * A section label in the [OdsModalDrawer] content.
+ *
+ * @property label Label of the section header.
+ */
+data class OdsModalDrawerSectionLabel(private val label: String) : OdsModalDrawerItem() {
```

Comment:
Should we rename it to `OdsModalDrawerSectionHeader`? Because in the documentation we talk about "section header".